### DISCUSSION TITLE
Planning

### DISCUSSION THREAD

**Discussion by maidenone** [maidenone/RobotDescriptor#16] [General]
Planning thread

**Comment by maidenone** [comment]
> so this is what I have in mind
> 
> 1. selecting the assembly workflow to import models from e.g assembly 4, a2plus , or the integrated workbench
> 2. create a QtTreeview to allow editing model (link and joint properties)
>    something like this
>    ![ign_robot_tree](https://private-user-images.githubusercontent.com/99465143/293168756-23548999-0e15-4c49-abb1-7694e435880c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3NTM0MDksIm5iZiI6MTcwMzc1MzEwOSwicGF0aCI6Ii85OTQ2NTE0My8yOTMxNjg3NTYtMjM1NDg5OTktMGUxNS00YzQ5LWFiYjEtNzY5NGU0MzU4ODBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI4VDA4NDUwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkMTBhZjA4NmQzNGYzNDlhNmU3Y2RlMDVlYmY2MzYwZGFkYjc4N2FhMDYwOWY0MDViZjA0OWQ1MTkxZmI2NzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oIJ5QiA0LmEQBIHuQ3J54IePnKhiBpJC09uMhPSpytI)
>    this is an screenshot from ignition
>    where selecting a link or joint would display the properties to allow a user to edit them
> 3. A way to add plugins depending on what it added to the element tree
> 4. a way to add sensors
> 5. Mechanisms to detect simple shapes such as boxes and cylinders , basically objects with internal name box,cylinder ...
> 6. *Maybe at a later time a way to import models from the arch workbench for something like warehouse setups

**Reply by maidenone** [reply]
This sounds like the direction i was heading with RobotCreator.

Can you expand on 5.? what do you want to do with the shapes?

**Comment by Jsevenh** [comment]
sure , for such shapes no need to generate meshes ,  as sdf and urdf already  supports those so instead of generating meshes , just read the  length width and height, or  radius and height for cylinders  and create a box element to represent them

**Comment by Jsevenh** [comment]
I forgot about generating a launch file to accompany the  exported model

**Comment by maidenone** [comment]
Right, that makes sense! that is a good optimization, and might actually not be that hard as they are "special" objects in FreeCAD.

**Reply by Jsevenh** [reply]
Yep I totally agree,  just reading the internal names and checking if its one of those shapes

**Comment by Jsevenh** [comment]
so  about the plan , starting with the assembly workbench would be a good start, I've downloaded the[ ondsel build ](https://ondsel.com/download/) of freecad to try the assembly workbench, what's your take on the assembly import part?

**Comment by maidenone** [comment]
Have mainly used LinkStage3/Asm3, Have just watched https://youtu.be/MI2JMm__bkM?si=jlSQqju16-F8zJrc and it looks great :) would be nice to support the joint format as that would solve some issues for us. But i do not think the assemby joints and SDF joints will map 1:1 so we will have to modify/change some assembly joints.

**Reply by Jsevenh** [reply]
> But i do not think the assembly joints and SDF joints will map 1:1 so we will have to modify/change some assembly joints.

1.  yeah I think that will be one of the considerations that will need to look at ( this needs to be one  of the tasks to find a way to map the joints from assembly to sdf )

I think the integrated workbench  is an awesome choice considering It'll be  a part of freeCAD so no worries about support for multiple workbenches , and the  fact that it also adds local coordinate systems is awesome these can be used to create reference frames

**Comment by Jsevenh** [comment]
been trying the ondsel  assembly workbench ,it might be a good option , a bit buggy though

**Comment by Jsevenh** [comment]
Hello, great new year

**Comment by Jsevenh** [comment]
I suggest we go we assembly4 for now , as we wait for the next  FreeCAD release hopefully It will be released with the assembly workbench
